### PR TITLE
Test VS Code Version Matrix

### PR DESCRIPTION
**Pull request by jpogran**
Change default `npm test` behavior to use the version of VS Code specified in the `VSCODE_VER` environment variable, instead of testing hardcoded versions inside `runtest.ts`.

`VSCODE_VER` supports latest, insiders and version numbers lik 1.55.0.

If the `VSCODE_VER` environment variable is not specified, then the latest stable version of VS Code is automatically used.

### CODE REVIEW

**Comment by jpogran** [comment]
Good news, we now have separate runs for each platform and each vs code version we currently test against. 

Bad news, have uncovered an actual problem with vs code 1.55.0

**Comment by jpogran** [comment]
1.58.2 works

**Comment by jpogran** [comment]
VS Code 1.55 was released in March 2021. 1.58 was release in June 2021. A small subset of our users are still on 1.55 or lower.

The test that is erroring is an autocompletion expectation. The rest of the extension tests out ok.

As a result of all of the above, I pinned to 1.58.2 as there are no user reports from 1.55 users.

File: src/test/runTest.ts [hunk 1]
```diff
@@ -41,15 +41,24 @@ async function main(): Promise<void> {
...
     // start in the fixtures folder to prevent the language server from walking all the
     // project root folders, like node_modules
-    console.log('_______________LATEST_____________________');
-    await runTests(options);
-
-    console.log('_______________INSIDERS___________________');
-    options.version = 'insiders';
-    await runTests(options);
+    const vscodeVersion = process.env['VSCODE_VERSION'];
+    switch (vscodeVersion) {
```
Comment (dbanck) [inline src/test/runTest.ts:45]: What do you think of removing this `switch` statement entirely? With the new matrix, it's clear which version is being tested. We could then add `version` to the `TestOptions` object in line 34.

If we want to keep some output, we could do something like this:
```ts
console.log(`_______________${options.version ?? 'latest'}_____________________`);
```

**Review by dbanck — APPROVED** [review]
LGTM! Just added a suggestion to get rid of some code, but I'm also fine if you want to keep it.

**Review by dbanck — APPROVED** [review]
Nice work!

**Pull request merged**